### PR TITLE
[action] mark access list tx in Execution

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
User could send an access list tx with empty access list, so we need to use the type of the Ethereum tx `tx.Type()`, but not `len(tx.accessList)` to determine if a tx is access list tx or not

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/builder.go [hunk 1]
```diff
@@ -181,7 +181,15 @@ func getRecipientAddr(addr *common.Address) string {
 // BuildExecution loads executino action into envelope
 func (b *EnvelopeBuilder) BuildExecution(tx *types.Transaction) (Envelope, error) {
 	b.setEnvelopeCommonFields(tx)
-	exec, err := NewExecutionWithAccessList(getRecipientAddr(tx.To()), tx.Nonce(), tx.Value(), tx.Gas(), tx.GasPrice(), tx.Data(), tx.AccessList())
+	var (
+		exec *Execution
+		err  error
+	)
+	if tx.Type() == types.AccessListTxType {
```
Comment (dustinxie) [inline action/builder.go:188]: need to use `tx.Type()` to determine if it's access list tx or not

File: action/protocol/execution/protocol_test.go [hunk 2]
```diff
@@ -270,15 +270,29 @@ func readExecution(
...
-		state.PendingNonce(),
-		ecfg.Amount(),
-		ecfg.GasLimit(),
-		ecfg.GasPrice(),
-		ecfg.ByteCode(),
-		ecfg.AccessList(),
+	var (
+		exec *action.Execution
 	)
+	if sct.InitGenesis.IsLondon || sct.InitGenesis.IsShanghai {
```
Comment (dustinxie) [inline action/protocol/execution/protocol_test.go:276]: supported in London and Shanghai EVM

File: action/rlp_tx_test.go [hunk 3]
```diff
@@ -411,6 +423,21 @@ var (
...
+			"1000000000000",
+			"0",
+			"0xBBd4e6E5dA2553a58a1E6b114E68cd37974e87a9",
+			4690,
+			iotextypes.Encoding_ETHEREUM_ACCESSLIST,
+			4,
+			"bc2d6c7699d0e3fa4f776b9acec2aaa37cf9f2c0f553db68caba5709ae05a149",
+			"04d92e345a51eb99efa71df0b452fa11246fd30516f4ee548f4edf999a3fb0e089993e58a2f137646911a89efd33100ec550ac535aa658f22f468af96583d06081",
+			"c592e5b215d639ed7889b5eaae1f98f9d09c9c86",
+		},
```
Comment (dustinxie) [inline action/rlp_tx_test.go:440]: add a test case with empty access list

**Review by envestcc — APPROVED** [review]

File: action/execution.go [hunk 4]
```diff
@@ -43,6 +43,7 @@ type Execution struct {
 	amount     *big.Int
 	data       []byte
 	accessList types.AccessList
+	isACL      bool
```
[the remark this answers is not in the document]
Comment (Liuhaai) [reply action/execution.go:46]: agree

[the remark this answers is not in the document]
Comment (dustinxie) [reply action/execution.go:46]: blob tx can carry a sidecar struct (containing blobs, commitments and proofs) which is very different. we'll have a new struct for that

[the remark this answers is not in the document]
Comment (CoderZhi) [reply action/execution.go:46]: I am also not very confident of this PR because of this modification. shall we leave it to next release such that we can have more discussion time?
